### DISCUSSION TITLE
`podman` without HOME

### DISCUSSION THREAD

**Discussion by abitrolly** [containers/podman#23144] [Q&A]
Is it possible to execute `podman` without HOME?

Looks like HOME is a hard requirement even for `--version` flag.

```sh
#!/usr/bin/env testscript

exec podman --version
```

Running `podman` with testscript fails.

```sh
#!/usr/bin/env testscript (0.028s)
> exec podman --version
[stderr]
time="2024-06-30T19:57:53+03:00" level=error msg="cannot resolve /no-home: lstat /no-home: no such file or directory"
[exit status 1]
FAIL: /tmp/testscript3074441362/dockerfile.txtar/script.txtar:3: unexpected command failure
error running ./dockerfile.txtar in /tmp/testscript3074441362/dockerfile.txtar
```

**Comment by Luap99** [comment]
I don't think it is possible, podman needs to parse config files before it sets up the cli commands/flags so there is no way around that even for something like the --version flag.

**Reply by abitrolly** [reply]
No config files == use defaults, no?

**Reply by Luap99** [reply]
yes but there is more initial setup magic happen, i.e. out code tries to set certain env very early on.

Do we need $HOME for podman --version, theoretically not. But the way the code is structures the core init logic runs always and it is very difficult to change that for just one command.

**Reply by abitrolly** [reply]
Can it `fail-on-use` instead instead of `fail-on-everything`?
